app/graphql/mutations/destroy_record.rb

```diff
@@ -2,35 +2,46 @@
 
 module Mutations
   class DestroyRecord < BaseMutation
-    argument :id, Integer, required: true
+    argument :id, Integer, required: false
     argument :record_type, String, required: true
+    argument :external_id, Integer, required: false
 
     type Types::DestroyType
 
     RECORD_WHITELIST = ["EventRecord", "NewsItem", "PointOfInterest", "Tour"].freeze
 
-    def resolve(id:, record_type:)
+    def resolve(id: nil, record_type:, external_id: nil)
       return error_status unless RECORD_WHITELIST.include?(record_type)
 
-      record = record_type.constantize
-                 .filtered_for_current_user(context[:current_user])
-                 .where(id: id)
-                 .first
+      record = if external_id.present?
+                 find_record(record_type, external_id: external_id)
+               else
+                 find_record(record_type, id: id)
```

Comment:
wenn keine `external_id` UND keine `id` gibt funktioniert `find_record` mit `id: nil`? kommt dann ein leeres `[]` und dann würde in `destroy_record` das present nicht greifen und `{ status: "Record not found", status_code: 404 }` resultieren?

Reply:
Guter Punkt. Eben nicht, da sollte dann noch ne guard clause rein. Am besten ich sicher dass alles nochmal durch nen Test ab.

Reply:
Ja, das ist richtig. When id nil ist dann returned find record nil und darauf returned destroy_record dann die Fehlermeldung